toolchain/lexer/token_kind.def

```diff
@@ -156,6 +156,7 @@ CARBON_KEYWORD_TOKEN(Return,        "return")
 CARBON_KEYWORD_TOKEN(Returned,      "returned")
 CARBON_KEYWORD_TOKEN(SelfParameter, "self")
 CARBON_KEYWORD_TOKEN(SelfType,      "Self")
+CARBON_KEYWORD_TOKEN(StringTypeLiteral, "String")
```

Comment:
Can you add a TODO or FIXME to document that this is a temporary thing until we have a prelude working?

Reply:
> Can you add a TODO or FIXME to document that this is a temporary thing until we have a prelude working?

I'm adding a vague TODO per your request, but unless I'm misunderstanding the result of https://github.com/carbon-language/carbon-lang/issues/2113#issuecomment-1301589247, I don't think this is a "prelude working" problem... This is closer to the decision regarding `Type` versus `type` on #2360